logging.go

```diff
@@ -297,9 +300,11 @@ func isLogFileWritable(filename string) bool {
 		}
 	}
 
-	if _, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644); err != nil {
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
```

Comment:
I'm not sure if this is necessary / beneficial ... or if the golang garbage collector is smart enough with _, err to close the file by itself

Reply:
it is not, and definitely needed.

you can either `f.Close()` or `defer f.Close()` im fine either way.